src/riak_cs_utils.erl

```diff
@@ -435,14 +435,19 @@ update_obj_value(Obj, Value) when is_binary(Value) ->
 key_exists(RcPid, Bucket, Key) ->
     key_exists_handle_get_manifests(riak_cs_manifest:get_manifests(RcPid, Bucket, Key)).
 
--spec big_end_key(non_neg_integer()) -> binary().
-big_end_key(NumBytes) ->
-    MaxByte = <<255:8/integer>>,
-    iolist_to_binary([MaxByte || _ <- lists:seq(1, NumBytes)]).
 
 -spec big_end_key() -> binary().
 big_end_key() ->
-    big_end_key(128).
+    big_end_key(<<>>).
+
+-spec big_end_key(Prefix::binary() | undefined) -> binary().
+big_end_key(undefined) ->
+    big_end_key(<<>>);
+big_end_key(Prefix) when byte_size(Prefix) > ?MAX_S3_KEY_LENGTH ->
```

Comment:
It's difficult to guess the "right" (not too bad) behavior in this situation...
The save start key and end one results in at most single entry. Then, is it better to use `<<Prefix/binary, 255>>` here?